### DISCUSSION TITLE
How to make use of query params when hosting the app?

### DISCUSSION THREAD

**Discussion by fahiar10** [Cinnamon/kotaemon#489] [Q&A]
I would like to pass query parameters to the application after running `python app.py`.

For example, When I visit http://127.0.0.1:7863/?param1=hello&param2=world, I want to be able to use these parameters and display them somewhere on the page.

Can someone guide me where to start?
Should I create fastAPIs to handle this?

https://huggingface.co/spaces/radames/gradio-url-params-request/blob/main/app.py
I tried with the gradio example above, but unable to integrate it with kotaemon.

**Comment by taprosoft** [comment]
Check out this issue https://github.com/gradio-app/gradio/issues/1608.
For BE API it is currently discussed at https://github.com/Cinnamon/kotaemon/discussions/330

**Reply by fahiar10** [reply]
Thank you for the response. I'm able to run the app with a custom endpoint using fastapi based on the thread given in your answer.
 
But my goal is to achieve the following after running the application,
When I go to the url,
http://127.0.0.1:7863/?prompt=hi+how+are+you

It should load the page and additionally the textbox for the chat window must be loaded from the url params.
Like so, 
![new](https://github.com/user-attachments/assets/f9aa11c8-1f58-4227-8761-beb9e117ab6f)


I found the part of the code where I can hardcode this (in libs/ktem/ktem/pages/chat/chat_panel.py), but ideally it should come from the url
![image](https://github.com/user-attachments/assets/c00534d3-3f64-4124-9aff-59a3f798e9dc)

Is there any pointers you could give me?
I have tried passing the params, but I think once the `demo = app.make()` is executed I can't make any changes to the ui while the app is running.
Also, the complicated inheritance structure is making it difficult for me understand on how to pass the query params from file to another. 🙃
